README.md

```diff
@@ -4,57 +4,120 @@ This is a tiny little shim around Cosmos SDK binaries that use the upgrade
+ It will return stdout and stderr of the subprocess as
+it's own. Because of that, it cannot accept any command line arguments, nor
+print anything to output (unless it dies before executing a binary).
+
+Configuration will be passed in the followingenvironmental variables:
+
+* `DAEMON_HOME` is the location where upgrade binaries should be kept (can
 be `$HOME/.gaiad` or `$HOME/.xrnd`)
-* we use [go-getter](https://github.com/hashicorp/go-getter) for all URI's which
-includes the ability to automatically compute SHA256 checksum's and unpack archives
-* place the genesis binary at `$DAEMON_HOME/upgrade_manager/genesis` or point
-`GENESIS_BINARY` to a [go-getter](https://github.com/hashicorp/go-getter) URI
-to retrieve it from
-* place the binary for each upgrade at `$DAEMON_HOME/upgrade_manager/<name>`
-where `<name>` is the URI-encoded name of the upgrade as specified in the upgrade
-module plan
-* or, store an os/architecture -> binary URI map in the upgrade plan info field
-as JSON (or YAML??) under the `"binaries"` key, eg:
-```json
-{
-  "binaries": {
-    "linux/amd64":"https://example.com/gaiad?checksum=sha256:b7d96c89d09d9e204f5fedc4d5d55b21"
-  }
-}
+* `DAEMON_NAME` is the name of the binary itself (eg. `xrnd`, `gaiad`)
+* `DAEMON_DOWNLOAD` (optional) if set to `on` will enable auto-downloading of new binaries
+(for security reasons, this is intended for fullnodes rather than validators)
+
+## Folder Layout
+
+`$DAEMON_HOME/upgrade_manager` is expected to belong completely to the upgrade manager and subprocesses
+constrolled by it. Under this folder, we will see the following:
+
 ```
-* or, set the upgrade plan to URI that points to a YAML or JSON file with the above structure 
-that can be retrieved by [go-getter](https://github.com/hashicorp/go-getter) 
-* all arguments passed to the upgrade manager command will be passed to the
-current daemon binary
-* the upgrade manager will monitor the stdout of the daemon to look for signals
-from the upgrade module indicating a pending or required upgrade and act
-appropriately, preferring on-disk binaries when they are present and defaulting
-to binaries provided in the on-chain upgrade plan when they are not
+- bin
+  - upgrade_manager (this binary)
+- genesis
+  - bin
+    - $DAEMON_NAME
+- upgrades
+  - <name>
+    - bin
+      - $DAEMON_NAME
+- current -> upgrades/foo
+```
+
+Each version of the chain is stored under either `genesis` or `upgrades/<name>`, which holds `bin/$DAEMON_NAME`
+along with any other needed files (maybe the cli client? maybe some dlls?). `current` is a symlink to the currently
+active folder (so `current/bin/$DAEMON_NAME` is the binary)
 
+Note: the `<name>` after `upgrades` is the URI-encoded name of the upgrade as specified in the upgrade module plan.
 
-## Layout of `DAEMON_HOME` Folder
+**Question** where is the state of the blockchain (config, data) stored? Under the update manager, or elsewhere?
```

Comment:
`$DAEMON_HOME` is usually just`~/.gaiad` or `~/.xrnd` so `$DAEMON_HOME/config` and `$DAEMON_HOME/data`. In the future we may use this knowledge to look up `$DAEMON_HOME/config/upgrade.json` or something

Reply:
We should either make this a standard, or just a recommendation.
But either way, make clear a possible connection there